### PR TITLE
Merge rackops_rolebook into platformstack

### PR DESCRIPTION
**Pull request by martinb3**
- Add `conflicts` against rackops_rolebook
- Copy all recipes from rackops_rolebook into platformstack. The only naming conflict was the default recipe, which was named `rackops_rolebook` when moved.
- Add library with a method for attribute access compatibility. Attributes previously in the `node['rackops_rolebook']` namespace should now be used at `node['platformstack']['rackops']`. The library will generate a warning when the older namespace is used.
- Copy all tests in from rackops_rolebook, and make them compatible with recent serverspec/chefspec.
- Update documentation to explain what happened.

### CODE REVIEW

**Comment by martinb3** [comment]
retest this please

**Comment by martinb3** [comment]
retest this please

File: libraries/platformstack.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+    deprecated_key = ['rackops_rolebook', keys].flatten
+    deprecated_value = current_node.deep_fetch(*deprecated_key)
+
+    current_key = ['platformstack', 'rackops', keys].flatten
+    current_value = current_node.deep_fetch(*current_key)
+
+    if deprecated_value && !current_value
+      Chef::Log.warn("You are using deprecated attribute keys under node['rackops_rolebook']: #{keys.join(',')}")
+    end
```
Comment (jujugrrr) [inline libraries/platformstack.rb:16]: Should we add a warning if we have both value set as well?

**Comment by jujugrrr** [comment]
It looks good, the change was not that big after all. Definitely a broker though, so we all have to be careful we don't upgrade to Platformstack >= 2.0.0 without actually wanting it.

File: metadata.rb [hunk 2]
```diff
@@ -2,35 +2,40 @@
...
+depends 'git'
+depends 'motd-tail', '~> 2.0'
+depends 'ohai', '~> 2.0'
+depends 'rackspace_iptables', '~> 1.7'
+depends 'sudo', '~> 2.7'
+depends 'user', '~> 0.3'
+depends 'chef-sugar'
+
+# conflicts with any version @ 2.0.0 of this cookbook
+conflicts 'rackops_rolebook'
```
Comment (prometheanfire) [inline metadata.rb:40]: can we concat and alphabetized the depends here?

File: recipes/default.rb [hunk 3]
```diff
@@ -19,6 +19,8 @@
 # limitations under the License.
 #
 
+include_recipe 'chef-sugar'
```
Comment (prometheanfire) [inline recipes/default.rb:22]: why is this needed?

Comment (martinb3) [inline recipes/default.rb:22]: The `Platformstack` library uses `node#deep_fetch`, so the gem must be installed before any calls to the library.

**Comment by marcoamorales** [comment]
:+1:

**Comment by jujugrrr** [comment]
Feel free to merge when everybody's happy :-). I'm :smile:
